### PR TITLE
Remove Dataflow hack from shared translation code

### PR DESCRIPTION
**Pull request by kennknowles**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
run dataflow validatesrunner

File: runners/google-cloud-dataflow-java/pom.xml [hunk 1]
```diff
@@ -199,6 +199,7 @@
                   <includes>
                     <include>com.google.guava:guava</include>
                     <include>com.google.protobuf:protobuf-java</include>
```
Comment (lukecwik) [inline runners/google-cloud-dataflow-java/pom.xml]: sdks/java/core, runners/direct-java should be updated as well if it was found that protobuf-java depends on having protobuf-java-util

Also, update gradles shade configurations.

Comment (kennknowles) [reply runners/google-cloud-dataflow-java/pom.xml]: Done. They aren't exactly consistent, so I kept each thing internally consistent. Wherever protobuf is shaded, I shaded protobuf-util.

To be clear, protobuf-java-util is used by `WindowingStrategyTranslation`, so it is places that use both protobuf-java-util and also translation aka runners/core-construction-java. So we don't always have to shade them both together.

**Comment by kennknowles** [comment]
Looks like the additional shading throughout broke the example ITs.

**Comment by kennknowles** [comment]
But suspiciously did not break the same way in gradle.

**Comment by kennknowles** [comment]
run dataflow validatesrunner

**Comment by kennknowles** [comment]
So far, I have not been able to reproduce a failure. Wondering if it has to do with either the one-command build or the all-the-runners profile, since those would also be different for the gradle build.

**Comment by kennknowles** [comment]
run java gradle precommit

**Comment by kennknowles** [comment]
retest this please

**Comment by kennknowles** [comment]
Confirmed by looking at other PRs that the failure in Dataflow WordCountIT is present on `master` (!!)

**Comment by kennknowles** [comment]
Diagnosed the issue - it is a Dataflow limitation that we are hitting across the board. Gonna merge this fix.

**Comment by kennknowles** [comment]
It is https://issues.apache.org/jira/browse/BEAM-3397.

**Pull request merged**